### PR TITLE
build: Use k8s v1.24 for tests

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

File: charts/kubefed/templates/federatedtypeconfig.yaml
Comment (jimmidyson) [inline charts/kubefed/templates/federatedtypeconfig.yaml:61]: `extensions/v1beta1.Ingress` was deprecated in k8s v1.14 and `networking.k8s.io/v1.Ingress` is available from k8s v1.19 so kubefed will still support ingresses back to v1.19 - this is an acceptable support window considering v1.24 is now out.

**Comment by jimmidyson** [comment]
Splitting into multiple PRs for easier review.

File: charts/kubefed/templates/federatedtypeconfig.yaml
Comment (makkes) [reply charts/kubefed/templates/federatedtypeconfig.yaml:61]: We should make sure to properly communicate this breaking change.

File: scripts/download-binaries.sh
Comment (makkes) [inline scripts/download-binaries.sh]: Is this missing a `$` sign somewhere?

File: scripts/download-binaries.sh
Comment (makkes) [inline scripts/download-binaries.sh:48]: Why do we need to setup envtest here?

File: charts/kubefed/templates/federatedtypeconfig.yaml
Comment (jimmidyson) [reply charts/kubefed/templates/federatedtypeconfig.yaml]: I'm not sure if this is a breaking change :thinking: Anyone using a newer version of Kubernetes will not be able to propagate ingress without this change... the breaking change is in the deletion of the API in Kubernetes, this is actually a fix!

Comment (jimmidyson) [reply charts/kubefed/templates/federatedtypeconfig.yaml:61]: Pulled this change out into #1521 to keep the PRs small and focused.

**Comment by jimmidyson** [comment]
/hold

Blocking until #1521 is merged.

File: charts/kubefed/templates/federatedtypeconfig.yaml
Comment (irfanurrehman) [reply charts/kubefed/templates/federatedtypeconfig.yaml:61]: I tend to agree with @jimmidyson. This is not a breaking change, but may be worth documenting anyways. @jimmidyson you also mentioned that you will cut a new release post this. It should ideally be sufficient to explicitly point this out in the release notes.

File: test/common/fixtures/ingresses.networking.k8s.io.yaml
Comment (irfanurrehman) [inline test/common/fixtures/ingresses.networking.k8s.io.yaml:1]: perhaps a rebase needed given https://github.com/kubernetes-sigs/kubefed/pull/1521 is merged.

Comment (jimmidyson) [reply test/common/fixtures/ingresses.networking.k8s.io.yaml:1]: :+1: that's why i blocked it, thanks for reminding me.

Comment (jimmidyson) [reply test/common/fixtures/ingresses.networking.k8s.io.yaml]: Done.

**Review by makkes — APPROVED** [review]

File: charts/kubefed/templates/federatedtypeconfig.yaml
Comment (jimmidyson) [reply charts/kubefed/templates/federatedtypeconfig.yaml]: @irfanurrehman I'll make sure to do that.

**Pull request merged**
